File: problems/0049-group-anagrams/python/solution.py
def groupAnagrams(strs):
    """Group Anagrams — https://leetcode.com/problems/group-anagrams/"""
    groups = {}
    for s in strs:
        key = "".join(sorted(s))
        groups.setdefault(key, []).append(s)
    return list(groups.values())

File: problems/0049-group-anagrams/typescript/solution.ts
// Group Anagrams — https://leetcode.com/problems/group-anagrams/
export function groupAnagrams(strs: string[]): string[][] {
  const groups = new Map<string, string[]>();
  for (const s of strs) {
    const key = [...s].sort().join("");
    const group = groups.get(key);
    if (group) group.push(s);
    else groups.set(key, [s]);
  }
  return [...groups.values()];
}
